context/AuthContext.tsx

```
@@ -0,0 +1,59 @@
+}>({ state: initialState, dispatch: () => null });
+
+const reducer = (state: AuthState, action: AuthAction): AuthState => {
+  switch (action.type) {
+    case "LOGIN":
+      localStorage.setItem("user", JSON.stringify(action.payload));
+      return { user: action.payload };
+    case "LOGOUT":
+      localStorage.removeItem("user");
+      return { user: null };
+    case "UPDATE_USER":
+      const updatedUser = { ...state.user!, ...action.payload };
```

Biome (1.9.4)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Unsafe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)